src/common/libjob/job.h

```diff
@@ -99,12 +111,17 @@ int flux_job_wait_get_id (flux_future_t *f, flux_jobid_t *id);
  * states can be set to an OR of any job state or any virtual job
  * states to retrieve jobs of only those states.  Specify 0 for all
  * states.
+ *
+ * results can be set to an OR of any job result to retrieve only
+ * those results.  Requires FLUX_JOB_INACTIVE states to retrieved.
+ * Specify 0 for all results.
  */
 flux_future_t *flux_job_list (flux_t *h,
                               int max_entries,
                               const char *json_str,
                               uint32_t userid,
-                              int states);
+                              int states,
+                              int results);
```

Comment:
I don't really have any alternatives for now, but we should think about a better way to make the `flux_job_list()` function extensible. Once deployed, we can't just add extra arguments to the function for each new thing we want to query or filter.

I kind of wonder if we even need this C interface on top of a more flexible JSON RPC mechanism, which is much more extensible without breaking all users. I'd hate to be limited in what we could do for job query by a C interface.

My only point for now is I wonder if it makes more sense to allow the `flux_job_list()` interface to query only states and not results. Since Python doesn't call the C interface, this wouldn't affect flux-jobs functionality.

Reply:
> My only point for now is I wonder if it makes more sense to allow the flux_job_list() interface to query only states and not results. Since Python doesn't call the C interface, this wouldn't affect flux-jobs functionality.

Seems like a good idea until we can get a more extensible plan in place.

Reply:
oh wait, then this eliminates my tests in `t/t2204-job-info` which is not great since we aren't adding a `--results` option to `flux-jobs`.  

Hmmm, I guess I could just build the rpc by hand via `t/request/rpc`.

Reply:
> Hmmm, I guess I could just build the rpc by hand via t/request/rpc.

Yup, this will work (although I gotta hardcode some bitmask values)